<launch>

  <!-- <arg name="map_yaml"                                                   doc="filepath to previously captured map (required)" /> -->
  <arg name="rviz"                   default="true"                      doc="whether to show Rviz" />
  <arg name="gazebo_world"           default="worlds/willowgarage.world" doc="the environment within which Stretch is loaded in Gazebo" />
  <arg name="gazebo_gpu_lidar"       default="false"                     doc="whether to compute lidar with hardware acceleration (requires GPU)" />
  <arg name="gazebo_visualize_lidar" default="false"                     doc="whether to visualize planar lidar within Gazebo" />

  

  <!-- MAP SERVER -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(find alfred_navigation)/maps/nursing_home_map.yaml" />

  <!-- LOCALIZATION -->
  <include file="$(find alfred_navigation)/launch/navigation/amcl_sim.launch" />

  <!-- NAVIGATION -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find alfred_navigation)/config/sim/common_costmap_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find alfred_navigation)/config/sim/common_costmap_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find alfred_navigation)/config/sim/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find alfred_navigation)/config/sim/global_costmap_params_withmap.yaml" command="load" />
    <!-- <rosparam file="$(find alfred_navigation)/config/sim/dwa_local_planner_params.yaml" command="load" /> -->
    <rosparam file="$(find alfred_navigation)/config/sim/base_local_planner_params.yaml" command="load" />
    <rosparam file="$(find alfred_navigation)/config/sim/move_base_params.yaml" command="load" />
    <remap from="/cmd_vel" to="/stretch_diff_drive_controller/cmd_vel" />
  </node>

  <!-- VISUALIZE -->
  <node name="rviz" pkg="rviz" type="rviz" output="log" args="-d $(find alfred_navigation)/rviz/navigation.rviz" if="$(arg rviz)" />

</launch>
